controllers/subscription_controller.go

```diff
@@ -290,12 +300,25 @@ func (r *SubscriptionReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	)
 
+	enqueueOdfSub := handler.EnqueueRequestsFromMapFunc(
+		func(ctx context.Context, obj client.Object) []reconcile.Request {
+			logger := log.FromContext(ctx)
+			sub, err := GetOdfSubscription(r.Client)
```

Comment:
We should not be running client command inside an Enqueue or request or a predicate. 
The only 2 ways we are allowed to invoke is: 
- Cast the obj to the event's object then use information on it to get to the desired CR (maybe using an ownerref, or an annotation)
- Locate the object before setting defining setting the enqueue command and use that name as a constant within the enqueue function

If we need to load the name of a constant CR we should do that before setting the E

Reply:
I removed this and passing the constant name.